clang/docs/TypeSanitizer.rst

```diff
@@ -0,0 +1,153 @@
+================
+TypeSanitizer
+================
+
+.. contents::
+   :local:
+
+Introduction
+============
+
+TypeSanitizer is a detector for strict type aliasing violations. It consists of a compiler
+instrumentation module and a run-time library. The tool detects violations such as the use 
+of an illegally cast pointer, or misuse of a union.
```

Comment:
I think this should say something about accessing memory with a different type than the dynamic type of the object or something like that?